src/black/output.py

```diff
@@ -8,34 +8,40 @@
 from mypy_extensions import mypyc_attr
 import tempfile
 
-from click import echo, style
+from click import echo, style as _style
+
+from black.const import OutputLevels
 
 
 @mypyc_attr(patchable=True)
 def _out(message: Optional[str] = None, nl: bool = True, **styles: Any) -> None:
     if message is not None:
-        if "bold" not in styles:
-            styles["bold"] = True
-        message = style(message, **styles)
+        message = _style(message, **styles)
     echo(message, nl=nl, err=True)
 
 
 @mypyc_attr(patchable=True)
-def _err(message: Optional[str] = None, nl: bool = True, **styles: Any) -> None:
-    if message is not None:
-        if "fg" not in styles:
-            styles["fg"] = "red"
-        message = style(message, **styles)
-    echo(message, nl=nl, err=True)
+def out(
+    message: Optional[str] = None,
+    nl: bool = True,
+    style: OutputLevels = OutputLevels.null,
+    **_styles: Any,
+) -> None:
+    styles = style.value.copy()
+    if style:
+        styles.update(_styles)
 
-
-@mypyc_attr(patchable=True)
-def out(message: Optional[str] = None, nl: bool = True, **styles: Any) -> None:
     _out(message, nl=nl, **styles)
 
 
 def err(message: Optional[str] = None, nl: bool = True, **styles: Any) -> None:
-    _err(message, nl=nl, **styles)
+    if isinstance(styles, dict):
+        if not styles.get("fg"):
+            styles["fg"] = "red"
+    elif len(styles) == 0:
+        styles = OutputLevels.error.value
+
+    _out(message, nl, **styles)
```

Comment:
```suggestion
    return out(message, nl, OutputLevels.error, **styles)
```
I feel like this could be condensed a bunch as well. And surely we always want the error level style as a base.